src/khoj/app/settings.py

```diff
@@ -112,20 +114,95 @@
     },
 ]
 
-ASGI_APPLICATION = "app.asgi.application"
 
 CLOSE_CONNECTIONS_AFTER_REQUEST = True
 
 # Database
 # https://docs.djangoproject.com/en/4.2/ref/settings/#databases
 DATA_UPLOAD_MAX_NUMBER_FIELDS = 20000
+
+# Default PostgreSQL configuration
+DB_NAME = os.getenv("POSTGRES_DB", "khoj")
+DB_HOST = os.getenv("POSTGRES_HOST", "localhost")
+DB_PORT = os.getenv("POSTGRES_PORT", "5432")
+
+# Use pgserver if env var explicitly set to true
+USE_PGSERVER = is_env_var_true("POSTGRES_USE_PGSERVER")
+
+if USE_PGSERVER:
+    # Set up logging for pgserver
+    logger = logging.getLogger("pgserver_django")
+    logger.setLevel(logging.INFO)
+    if not logger.handlers:
+        handler = logging.StreamHandler()
+        handler.setFormatter(logging.Formatter("%(asctime)s - %(name)s - %(levelname)s - %(message)s"))
+        logger.addHandler(handler)
+
+    WSGI_APPLICATION = "app.wsgi.application"
+
+    try:
+        import pgserver
+
+        # Set up data directory
+        PGSERVER_DATA_DIR = os.path.join(BASE_DIR, "pgserver_data")
+        os.makedirs(PGSERVER_DATA_DIR, exist_ok=True)
+
+        logger.info(f"Initializing pgserver with data directory: {PGSERVER_DATA_DIR}")
+
+        # Start server
+        PGSERVER_INSTANCE = pgserver.get_server(PGSERVER_DATA_DIR)
+
+        # Create pgvector extension, if not already exists
+        PGSERVER_INSTANCE.psql("CREATE EXTENSION IF NOT EXISTS vector;")
```

Comment:
Do you know if pg_server comes with the `vector` extension pre-baked? We have it on our machines because it came with postgres.app on macos. Just a quick thing to verify. Maybe we can also distribute the extension in the Khoj app if needed(?).